### PR TITLE
feat(wallet): NFT Pinning design updates

### PR DESCRIPTION
**Pull request by muliswilliam**
Resolves https://github.com/brave/brave-browser/issues/28986
https://github.com/brave/brave-browser/issues/29009

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
Screenshot 2023-03-13 at 18 11 54
Screenshot 2023-03-13 at 18 12 11

### CODE REVIEW

File: components/brave_wallet_ui/components/desktop/nft-ipfs-banner/nft-ipfs-banner.style.ts [hunk 1]
```diff
@@ -24,13 +30,15 @@ export const StyledWrapper = styled.div<{ status: BannerStatus }>`
   justify-content: flex-start;
   background: ${(p) =>
     p.status === 'start'
-      ? `url(${BannerBackground}) right 80px center/contain no-repeat, linear-gradient(110.74deg, #242464 -10.97%, #000027 173.98%)`
+      ? `url(${BannerBackground}) right 80px center/contain no-repeat, #1A1C3B`
       : p.status === 'uploading'
-      ? '#F0F7FC'
+      ? getUploadingBackground()
       : getSuccessBackground()};
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/nft-ipfs-banner/nft-ipfs-banner.style.ts:36]:
```suggestion
--success-background: #E8F9EB;
--uploading-background: #F0F7FC;
@media (prefers-color-scheme: dark) {
    --success-background: #15371B;
    --uploading-background: #E8F9EB;
}
background: ${(p) =>
  p.status === 'start'
    ? `url(${BannerBackground}) right 80px center/contain no-repeat, #1A1C3B`
    : p.status === 'uploading'
    ? var(--uploading-background)
    : var(--success-background)};
```

File: components/brave_wallet_ui/components/desktop/nft-ipfs-banner/nft-ipfs-banner.style.ts [hunk 2]
```diff
@@ -12,8 +12,14 @@ import { BannerStatus } from './nft-ipfs-banner'
...
-    : '#EEFBF0'
+    ? '#15371B' // Leo Value: Dark/SystemFeedback/Success-Background
+    : '#E8F9EB' // Leo Value: Light/SystemFeedback/Success-Background
+}
+
+const getUploadingBackground = () => {
+  return window.matchMedia('(prefers-color-scheme: dark)').matches
+    ? '#063256' // Leo Value: Dark/SystemFeedback/Info-Background
+    : '#F0F7FC' // Leo Value: Light/SystemFeedback/Info-Background
 }
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/nft-ipfs-banner/nft-ipfs-banner.style.ts:23]: Can remove these!

File: components/brave_wallet_ui/components/desktop/nft-ipfs-banner/nft-ipfs-banner.style.ts [hunk 3]
```diff
@@ -40,7 +48,7 @@ export const Text = styled.p<{ status: BannerStatus }>`
   font-size: 14px;
   line-height: 20px;
   align-items: center;
-  color: ${(p) => (p.status === 'start' ? p.theme.palette.white : '#1D1F25')};
+  color: ${(p) => (p.status === 'start' ? p.theme.palette.white : window.matchMedia('(prefers-color-scheme: dark)').matches ? '#ECEFF2' : '#1D1F25')};
```
Comment (Douglashdaniel) [inline components/brave_wallet_ui/components/desktop/nft-ipfs-banner/nft-ipfs-banner.style.ts:51]:
```suggestion
--text-color: #1D1F25;
@media (prefers-color-scheme: dark) {
  --text-color: #ECEFF2;
}
color: ${(p) => p.status === 'start' ? p.theme.palette.white : var(--text-color)};
```

**Review by Douglashdaniel — CHANGES_REQUESTED** [review]

File: components/brave_wallet_ui/components/desktop/nft-ipfs-banner/nft-ipfs-banner.style.ts (hunk 2 above)
Comment (muliswilliam) [reply components/brave_wallet_ui/components/desktop/nft-ipfs-banner/nft-ipfs-banner.style.ts]: Great regactor. Done here 37843dd9ab7d9e3180cec056c996588accc674aa

**Review by Douglashdaniel — APPROVED** [review]
++

File: components/resources/wallet_strings.grdp [hunk 4]
```diff
@@ -587,7 +587,7 @@
   <message name="IDS_BRAVE_WALLET_SOLANA_ASSOCIATED_TOKEN_PROGRAM" desc="Name of Solana Associated Token Program">Associated Token Program</message>
   <message name="IDS_BRAVE_WALLET_SOLANA_META_DATA_PROGRAM" desc="Name of Solana MetaData Program">MetaData Program</message>
   <message name="IDS_BRAVE_WALLET_SOLANA_SYSVAR_RENT_PROGRAM" desc="Name of Solana SysvarRent Program">SysvarRent Program</message>
-  
+
```
Comment (cypt4) [inline components/resources/wallet_strings.grdp:590]: nit: unnecessary diff

File: components/brave_wallet_ui/stories/locale.ts [hunk 5]
```diff
@@ -842,29 +842,32 @@ provideStrings({
...
+  braveWalletNftPinningRunNodeDescription: 'IPFS is like a big library where people can store their files, including NFT data. It\'s designed to make data retrieval faster and more efficient, so that when you request data, you can get it from the closest library or the nearest location to you where someone else already has the copy of data. This means that your NFT data can be accessed quickly and easily, as long as your NFT data stays online.',
   braveWalletNftPinningCheckNftsButton: 'Check which NFTs of mine can be pinned',
-  braveWalletNftPinningBannerStart: 'Now you can run your IPFS and be part of web 3. Your NFT data will stay online forever and cannot be tampered with.',
+  braveWalletNftPinningBannerStart: 'Now you can run your IPFS and be part of web 3. Your NFT data will stay online forever.',
   braveWalletNftPinningBannerUploading: 'You\’re running IPFS node. File is being uploaded to IPFS.',
```
Comment (cypt4) [inline components/brave_wallet_ui/stories/locale.ts:857]: nit: this string doesn't reflect grdp content

**Review by cypt4 — APPROVED** [review]
LGTM

**Comment by srirambv** [comment]
Verification passed on

Brave | 1.52.17 Chromium: 112.0.5615.49 (Official Build) nightly (64-bit)
-- | --
Revision | `bd2a7bcb881c11e8cfe3078709382934e3916914-refs/branch-heads/5615@{#936}`
OS | Windows 11 Version 22H2 (Build 22621.1485)

<details>
<summary><b>Verified <code>https://github.com/brave/brave-browser/issues/28986</code> - Passed</b></summary>
<br>

- Verified UI updates to NFT pinning

Banner Main

Light | Dark
-- | --
brave_tuT7Bm5VMZ | brave_MdPuCWPAYz



Banner Uploading

Light | Dark
-- | --
brave_cyCKAfDNug | brave_5Z1OLXJKyy

Banner Success

Light | Dark
-- | --
brave_qf4NWwjD5s | brave_G8cqjvrWks

NFT-IPFS Interstitial page

Light | Dark 
-- | --
brave_vdgdqJKbb3 | brave_9GdezdpiIy


NFT-IPFS Interstitial page (Available NFT pins)

Light | Dark 
-- | --
brave_PcJrf46XWv | brave_9S64hbHT3L

NFT Details page

Light | Dark 
-- | --
brave_QnBnNMUTl9 | brave_PwLjGxxoxG

- Encountered [brave-browser#29514](https://github.com/brave/brave-browser/issues/29514)
</details>
<details>
<summary><b>Verified <code>https://github.com/brave/brave-browser/issues/29009</code> - Passed</b></summary>
<br>

- Verified NFT-IPFS Icon is always shown 
- Verified on hover changes the button colour
- Verified still able to click the button to show/hide the message
- Verified button visibility on both dark and light theme

https://user-images.githubusercontent.com/17010094/229984412-c427c9cc-70f2-417a-bead-aaef3fde0d4a.mp4

</details>

**Pull request merged**